Tests/UIKitTests/UIImageExtensionsTests.swift

```diff
@@ -13,6 +13,13 @@ import XCTest
 import UIKit
 
 final class UIImageExtensionsTests: XCTestCase {
+    func testAverageColor() {
+        // note that not all colors precisely survive the roundtrip due to CI colorspace
+        // management. Blue & brown seem safe, though.
```

Comment:
If you don't get exact values you can still use `XCTAssertEqualWithAccuracy` to check the components of the colours.
And how about adding, for example, an image with size (0, 0)? Or a combination of colours?

Reply:
ok, how about this? (see latest)